### PR TITLE
Feature: Added Boundary locator for Rectangular Shape

### PR DESCRIPTION
**Pull request by manevpe**
*Issue #, if available:*

*Description of changes:*
Added a new shape type Rectangle.
Added method for Boundary calculation for the intersection of rectangles.
Added a new demo with rectangular shape connectors with boundary setting.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/components/view/View.tsx [hunk 1]
```diff
@@ -156,6 +156,63 @@ const getBoundaryCoordinatesForCircle = (
   };
 };
 
+const getBoundaryCoordinatesForRectangle = (
```
Comment (sameergoyal) [inline src/components/view/View.tsx:159]: Can you add a unit test for this?

Comment (manevpe) [reply src/components/view/View.tsx:159]: I've added a couple of snapshot tests.

File: src/components/view/View.tsx [hunk 2]
```diff
@@ -156,6 +156,63 @@ const getBoundaryCoordinatesForCircle = (
...
+  sourcePosition: Position,
+  sourceSize: Size,
+  destinationPosition: Position
+): Position => {
+  // If the two points are orthogonal, the boundary is half the width or height of the rectangle.
+  if (sourcePosition.x === destinationPosition.x) {
+    return {
+      x: sourcePosition.x,
+      y: sourceSize.height / 2
```
Comment (sameergoyal) [inline src/components/view/View.tsx:168]: This should be sourcePosition.y + sourceSize.height if destinationPosition.y is greater than sourcePosition.y and sourcePosition.y if destionationPosition.y is less than sourcePosition.y. I am able to reproduce in your repo this bug when I bring a node exactly perpendicular either horizontally or vertically.

File: src/components/view/View.tsx [hunk 3]
```diff
@@ -156,6 +156,63 @@ const getBoundaryCoordinatesForCircle = (
...
+    };
+  }
+  if (sourcePosition.y === destinationPosition.y) {
+    return {
+      x: sourceSize.width / 2,
+      y: sourcePosition.y
```
Comment (sameergoyal) [inline src/components/view/View.tsx:174]: Same as above.

File: src/components/view/View.tsx [hunk 4]
```diff
@@ -156,6 +156,63 @@ const getBoundaryCoordinatesForCircle = (
...
+    };
+  }
+
+  // Otherwise - locate the boundary intersection point by calculating two right angle triangles.
+  // The first, bigger one, is created between the source and destination points.
+  // The second, smaller one, is created between the destination center and boundary intersection,
+  // with the rectangle. Use Pitagorean theorem for the calculations.
```
Comment (sameergoyal) [inline src/components/view/View.tsx:181]: Minor: Typo. Pythagorus theorem instead of Pitagorean theorem.

**Review by sameergoyal — CHANGES_REQUESTED** [review]
Apologies for the delay in reviewing. Was cross checking the calculations for coordinates and have found a bug. Please fix that and we should be good to merge.

And thanks for contributing to this library! :)

File: src/components/view/View.tsx (hunk 2 above)
Comment (sameergoyal) [reply src/components/view/View.tsx:168]: My bad. It should be sourcePosition.y + sourceSize.height/2 (if sourcePosition.y < destinationPosition.y) else sourcePosition.y - sourceSize.height/2 (if sourcePosition.y > destinationPosition.y)

**Comment by manevpe** [comment]
> Apologies for the delay in reviewing. Was cross checking the calculations for coordinates and have found a bug. Please fix that and we should be good to merge.
> 
> And thanks for contributing to this library! :)

No worries - it found a bug, so it was worth it :) 
I've commited a fix and added a new unit test for those cases.

File: src/components/view/View.tsx (hunk 1 above)
Comment (manevpe) [reply src/components/view/View.tsx:159]: Done

**Review by sameergoyal — APPROVED** [review]
Looks good to me after the changes.

Thanks for your contribution. We will add this into your next release.

**Pull request merged**